variables.tf

```diff
@@ -85,21 +85,22 @@ variable "kms_key_arn" {
   description = "The ARN for the KMS encryption key to encrypt the Redshift cluster"
 }
 
-variable "lifecycle_rule" {
-  type        = any
-  default     = []
-  description = "List of maps containing lifecycle management configuration settings"
-}
-
 variable "logging" {
-  type        = bool
-  default     = true
-  description = "Enables logging information such as queries and connection attempts"
-}
+  type = object({
+    bucket_name           = optional(string, null)
+    bucket_lifecycle_rule = optional(any, [])
+    create_bucket         = optional(bool, true)
+    log_destination_type  = string
+    log_exports           = optional(list(string), ["connectionlog", "useractivitylog", "userlog"])
+    s3_key_prefix         = optional(string, "redshift-audit-logs/")
```

Comment:
Can we keep this consistent with other bucket variables? Something like `bucket_key_prefix`?

Reply:
modified it